### PR TITLE
Revisions: Add notice with undo button

### PR DESCRIPTION
**Pull request by nheagy**
Refs #10304

This adds a notice with undo button when loading a revision.

revisions undo

**Note:** This currently also includes changes from #10633, and I'll wait until that is approved to merge.

To test:
- Load a revision from a posts's history
- Ensure a notice view appears
- Ensure that the undo button on the notice words

Update release notes:
- [x] I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by pinarol** [comment]
I'd like to point to the Gutenberg side of the things at this point. We have added History option recently to the Gutenberg editor, so any development about the History option(or other options that appear in the same action sheet like Preview, Blog settings etc.) will need testing with Gutenberg editor also. If test fails and the solution is somehow complex to handle within this PR please don't hesitate to create and assign an issue to me. The point is not losing the track of things and keeping aztec/gutenberg in sync.

File: WordPress/Classes/ViewRelated/Post/PostEditor+MoreOptions.swift [hunk 1]
```diff
@@ -22,10 +22,28 @@ extension PostEditor where Self: UIViewController {
 
     func displayHistory() {
         let revisionsViewController = RevisionsTableViewController(post: post) { [weak self] revision in
-            if let post = self?.post.update(from: revision) {
-                DispatchQueue.main.async {
-                    self?.post = post
+            guard let self = self else {
```
Comment (danielebogo) [inline WordPress/Classes/ViewRelated/Post/PostEditor+MoreOptions.swift]: I think unwrapping `self` here is not necessary and you can `guard let post = self.post.update(from: revision)` instead.

File: WordPress/Classes/ViewRelated/Post/PostEditor+MoreOptions.swift [hunk 2]
```diff
@@ -22,10 +22,28 @@ extension PostEditor where Self: UIViewController {
...
+                        return
+                    }
+                    let clone = self.post.clone(from: original)
+                    self.post = clone
+                }
+            }
+            ActionDispatcher.dispatch(NoticeAction.post(notice))
+
+            DispatchQueue.main.async {
+                self.post = post
```
Comment (danielebogo) [inline WordPress/Classes/ViewRelated/Post/PostEditor+MoreOptions.swift]: here `self` can be optional

File: WordPress/Classes/ViewRelated/Post/PostEditor+MoreOptions.swift [hunk 3]
```diff
@@ -22,10 +22,28 @@ extension PostEditor where Self: UIViewController {
...
+            let notice = Notice(title: "Revision loaded", message: nil, feedbackType: .success, notificationInfo: nil, actionTitle: "Undo", cancelTitle: nil) { [weak self] (happened) in
+                guard happened, let self = self else {
+                    return
                 }
+                DispatchQueue.main.async { [weak self] in
+                    guard let self = self, let original = self.post.original else {
+                        return
+                    }
+                    let clone = self.post.clone(from: original)
+                    self.post = clone
```
Comment (danielebogo) [inline WordPress/Classes/ViewRelated/Post/PostEditor+MoreOptions.swift]: Even here, no weak self needed and you can `guard let original` and `let clone`. `self.post` can also remain optional.

File: RELEASE-NOTES.txt [hunk 4]
```diff
@@ -1 +1,2 @@
-* App Settings Privacy: Added explanation for the 'Remove Location from Media' feature.
\ No newline at end of file
+* App Settings Privacy: Added explanation for the 'Remove Location from Media' feature.
+* Added a notice and undo button when loading a revision from a post's history.
```
Comment (jtreanor) [inline RELEASE-NOTES.txt]: Thank you! 👏

**Comment by danielebogo** [comment]
@nheagy @pinarol I tested it using Gutenberg and it works perfectly!

**Comment by nheagy** [comment]
@danielebogo cleaning up the references to `self` looks much nicer 👍  Care for another look?

**Comment by danielebogo** [comment]
@nheagy I tested it again (also with Gutenberg) and it works!
Nice work!
:shipit:

**Comment by nheagy** [comment]
Thanks @danielebogo!

**Pull request merged**